Fully integrated the outer GAN workflow

tgreed86 opened the pull request:
This pull request successfully includes the option to run either the inner GAN train or outer GAN training. 

- python3 ./aidapt_toolkit/drivers/hydra_driver.py to run the inner GAN training

- python3 ./aidapt_toolkit/drivers/outer_gan_driver.py to run the outer GAN training

The input yaml files are currently hard-coded in these drivers:

- hydra_basic_config.yaml for inner GAN

- hydra_outer_config.yaml for outer GAN

Additional diagnostics plots can be produced by specifying True or False in the "metrics" section in the yaml files. Additionally, the user can set the frequency (in terms of epoch) at which that these metrics will be calculated.

Commit:
Test commit

Commit:
Incorporating additional training monitoring metrics

Commit:
Updated code to run outer GAN. Also added ROC curve plot

Commit:
First commit for running outer GAN driver.

sgoldenCS committed:
Ensuring outerGAN train_step is called

This update seems to have fixed the initial problem where the outerGAN train_step() was never called. However, there is still a bug with getting the generator to predict properly (perhaps because it isn't "built"?). I will continue to work on a fix for the new bugs.

Commit:
Outer GAN now runs

Commit:
Fixed outer GAN generator bug

Commit:
Updates to outer GAN implementation.

Commit:
Fixed assortment of bugs, issues with scaling of data, and outer GAN functionality.

Commit:
Apply Black formatting

tgreed86 committed:
Merge branch 'main' into 12-bug-with-train_step-inheritance

sgoldenCS commented on Hall_B/AIDAPT/aidapt_toolkit/configs/testfile.txt:
@tgreed86, I believe this "testfile" can be removed, right?

sgoldenCS commented on Hall_B/AIDAPT/aidapt_toolkit/drivers/hydra_driver.py:
```diff
@@ -38,31 +39,65 @@ def run(config) -> None:
         config["v_scaler"]["id"], config=config["v_scaler"], name="vertex_scaler"
     )
+    v_s_scaler = aidapt_toolkit.data_prep.make(
+        config["v_scaler"]["id"], config=config["v_scaler"], name="vertex_scaler"
```
I don't think it matters when using Hydra, but maybe it would make sense to give this another name to avoid two "vertex_scaler"s? I'd have to spend some time remembering how the registry `make()` call and the workflow in general is set up, but I think we used the "name" parameter to avoid conflicts between two different objects with the same "ID".

sgoldenCS commented on Hall_B/AIDAPT/aidapt_toolkit/drivers/hydra_driver.py:
```diff
@@ -126,8 +187,8 @@ def run(config) -> None:
     vertex_parser.save(os.path.join(save_path, "vertex_parser"))
     detector_parser.save(os.path.join(save_path, "detector_parser"))
     lab2inv_prep.save(os.path.join(save_path, "lab2inv_prep"))
-    d_scaler.save(os.path.join(save_path, "d_scaler"))
-    v_scaler.save(os.path.join(save_path, "v_scaler"))
+    # d_scaler.save(os.path.join(save_path, "d_scaler"))
+    # v_scaler.save(os.path.join(save_path, "v_scaler"))
```
Is there a reason we no longer save the scalers? I would assume they would be helpful to have for offline analysis.

tgreed86 replied:
I probably did this at some point during testing and never uncommented them. I've removed the comments.

sgoldenCS commented on Hall_B/AIDAPT/aidapt_toolkit/drivers/outer_gan_driver.py:
```diff
@@ -1,27 +1,193 @@
+        config["v_scaler"]["id"], config=config["v_scaler"], name="vertex_scaler"
+    )
+    v_s_scaler = aidapt_toolkit.data_prep.make(
+        config["v_scaler"]["id"], config=config["v_scaler"], name="vertex_scaler"
```
Similar comment here about the "name" parameter compared to the inner_gan driver.

sgoldenCS reviewed (commented):
Overall things look pretty good! I just have a few small comments that I think could be addressed quickly and then we can complete the merge/PR.

tgreed86 committed:
Delete Hall_B/AIDAPT/aidapt_toolkit/configs/testfile.txt

tgreed86 committed:
Uncomment saving of scalers in hydra_driver.py

tgreed86 committed:
Changed name to "vertex_s_scaler" for corresponding scaler in hydra_driver.py

tgreed86 committed:
Changed name to "vertex_s_scaler" for corresponding scaler in outer_gan_driver.py

This is the final change that needed to be addressed. This branch should now be ready to merge to main.

sgoldenCS commented:
It looks like you've addressed my comments, so I will go ahead and merge it into the base branch. Thanks 👍
